### ISSUE TITLE
iws.mx can't be reached

### ISSUE THREAD

**Issue by Archel0n** [Sheep-y/trpg-dnd-4e-db#45]
Title. The database site can no longer be accessed. Even from a cached version of the website, it couldn't read information from the database.

In addition, Digital River's storefront for DDI subscriptions appears to have shut down, so there is no longer a way to purchase DDI subscriptions to gain offline access to the compendium.

I've been really enjoying using the database for my current 4e game, and my players have been using it for their characters as well, so I'd like to know what steps I should follow from here.

**Comment by KaiserGrizzly** [comment]
Facing the same issue here and wanted to add the same praises for the tool!

**Comment by mw-stanley** [comment]
You are correct that Dungeons & Dragons Insider is no longer available and subscriptions are no longer available for purchase.

I don't believe this repo or its maintainer has any relationship with whoever was hosting iws.mx. This repo was able to generate websites like iws.mx when DDI was available, but since it no longer is, this software no longer does anything, and the maintainers of this repo can't do anything about that. 

Hosting of sites like iws.mx was unlikely to be legal and it was probably subject to a takedown notice, and I would not expect the maintainers of this repo to be able to do anything to help with that either.

**Comment by Dreugui** [comment]
iws.mx is back ! 
but to avoid other surprise I would like to have a portable version. yes I downloaded the portable version  available in the other issue, but the search feature in it is really not eanough compared to the web version. and so is there a way to use the database of the portable version and use it as a source for the extractor to generate the portable version of the html ? 
EDIT : OK! so, I succeded to creat the html version but with NO DATA .(I think it's a bug ? but it's usefull for me so I don't care) if someone could link the folder with the data.js it would be life saving !

**Comment by Dreugui** [comment]
OMG ! I DID IT !
ALL the data are in the .js in iws.mx ! ( meaning you can save it on local EASILY  ! ). It took me some time to download all the js, 20js for each category. but it is DONE ! I have it ! 
of course it is usefull only if you have succeded to do the same as me and export the site without the data.
Anyway. good luck for thoose that come next

**Comment by Shananiganeer** [comment]
> OMG ! I DID IT ! ALL the data are in the .js in iws.mx ! ( meaning you can save it on local EASILY ! ). It took me some time to download all the js, 20js for each category. but it is DONE ! I have it ! of course it is usefull only if you have succeded to do the same as me and export the site without the data. Anyway. good luck for thoose that come next

How did you go about scraping the site? I'm an old former DDI subscriber that downloaded the compendium back in the day, but through a series of unfortunate events have lost that data and desperately want it back!

**Comment by Dreugui** [comment]
You juste have to type F12 when you're on the site. you can see ALL the JS on the page, you juste have to download them.
in the debug menu.
EDIT : so, it was long ago, and i just redone it :
 1) go on https://iws.mx/dnd/
 2) Open the debug menu of you browser
 3) click on "Everything"
 4) click on the first object (Reaping Disruption)
 5) press the arrow key right an WAIT to the end ( or long enough )
 6) on the debug menu you'll see ALL the JS beeing loaded.
 7) right click => register as files
 8) put the files in your local folder
 9) ENJOYs your DnD 4e local data base !

**Comment by Shananiganeer** [comment]
I ended up mirroring the site with wget, but it doesn't seem to work when I open the index file and the page comes up blank.

I also only saw a single JS file when I was peeking at it with the devtools which didn't line up with you downloading 20 per category.

EDIT: Nevermind, I see it now.

**Comment by Dreugui** [comment]
yea, they load only when needed so you HAVE to go through ALLLLLL the page :/

**Comment by Shananiganeer** [comment]
Ok, turns out iws.mx has directory browsing enabled, so you can just mirror the 4e_database_files directory and grab all files in one swoop.

wget -mnp -nH -e robots=off --reject-regex ".\*\\?.\*" https://iws.mx/dnd/4e_database_files/
wget -pnp -nH -e robots=off https://iws.mx/dnd/index.html

~The local version is still broken, but at least I have the data XD.~

Ok working now, had to remove the -k flag from wget as it was breaking the links.

The above wget commands will grab everything you need. Just move the resulting folders to your webserver of choice and navigate to the dnd/index.html file.
